components/brave_wallet_ui/common/async/handlers.ts

```diff
@@ -519,12 +519,15 @@ handler.on(WalletActions.refreshGasEstimates.type, async (store: Store, txInfo:
   const { ethTxManagerProxy, solanaTxManagerProxy } = getAPIProxy()
 
   if (isSolanaTransaction(txInfo)) {
-    const getSolFee = await solanaTxManagerProxy.getEstimatedTxFee(txInfo.id)
-    if (!getSolFee.fee) {
-      console.error('Failed to fetch SOL Fee estimates')
+    const { fee, errorMessage } = await solanaTxManagerProxy.getEstimatedTxFee(
+      txInfo.id
+    )
+    if (!fee) {
+      console.error('Failed to fetch SOL Fee estimates: ' + errorMessage)
+      store.dispatch(WalletActions.setHasFeeEstimatesError(true))
       return
     }
-    store.dispatch(WalletActions.setSolFeeEstimates({ fee: getSolFee.fee }))
+    store.dispatch(WalletActions.setSolFeeEstimates({ fee }))
```

Comment:
Just a remark not related to your PR -  I've always felt it's weird to store transaction-specific fees in a global state that's shared by everyone.